IPython/core/display.py

```diff
@@ -869,14 +870,14 @@ def _repr_html_(self):
             mimetype, encoding = mimetypes.guess_type(self.filename)
 
             video = open(self.filename, 'rb').read()
-            video_encoded = video.encode('base64')
+            video_encoded = base64.b64encode(video)
         else:
             video_encoded = self.data
             mimetype = self.mimetype
         output = """<video controls>
  <source src="data:{0};base64,{1}" type="{0}">
  Your browser does not support the video tag.
- </video>""".format(mimetype, video_encoded)
+ </video>""".format(mimetype, video_encoded.decode('ASCII'))
```

Comment:
I'm a little concern as this changes if `filename` is `None`, `video_encoded = self.data`, which suddenly get decoded as ASCII while it was not before.